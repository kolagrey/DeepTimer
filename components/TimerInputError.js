import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from '../styles';

const TimerInputError = (props) => {
    const { text } = props;
    return (
        <Text style={styles.redText}>
            {text}
        </Text>
    );
};

TimerInputError.propTypes = {
    text: PropTypes.string
};

export default TimerInputError;