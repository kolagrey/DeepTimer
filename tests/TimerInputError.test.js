import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import TimerInputError from '../components/TimerInputError';

describe('<TimerInputError/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<TimerInputError/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <TimerInputError />,
            ).toJSON();
        expect(tree.type).toBe('Text');
    });

    it('has style of color red', () => {
        const tree = renderer
            .create(
                <TimerInputError/>,
            ).toJSON();
        expect(tree.props.style.color).toBe('red')
    });
});