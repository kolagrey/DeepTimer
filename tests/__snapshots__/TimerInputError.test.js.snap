// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerInputError/> renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
